controllers/clusterresources/nodereload_controller.go

```diff
@@ -60,151 +68,188 @@ func (r *NodeReloadReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	if err != nil {
 		if k8serrors.IsNotFound(err) {
 			l.Error(err, "Node Reload resource is not found", "request", req)
-			return models.ExitReconcile, nil
+			return reconcile.Result{}, err
 		}
 		l.Error(err, "Unable to fetch Node Reload", "request", req)
-		return models.ReconcileRequeue, err
+		return reconcile.Result{}, err
 	}
 
-	if len(nrs.Spec.Nodes) == 0 {
-		err = r.Client.Delete(ctx, nrs)
+	patch := nrs.NewPatch()
+
+	if len(nrs.Status.PendingNodes)+len(nrs.Status.CompletedNodes)+len(nrs.Status.FailedNodes) == 0 {
+		nrs.Status.PendingNodes = nrs.Spec.Nodes
+		err := r.Status().Patch(ctx, nrs, patch)
 		if err != nil {
-			l.Error(err,
-				"Cannot delete Node Reload resource from K8s cluster",
-				"Node Reload spec", nrs.Spec,
+			l.Error(err, "Failed to patch pending nodes to the resource")
+			r.EventRecorder.Eventf(nrs, models.Warning, models.PatchFailed,
+				"Failed to patch pending nodes to the resource. Reason: %w", err,
 			)
-			r.EventRecorder.Eventf(
-				nrs, models.Warning, models.DeletionFailed,
-				"Resource deletion is failed. Reason: %v",
-				err,
-			)
-			return models.ReconcileRequeue, nil
+
+			return reconcile.Result{}, err
 		}
+	}
+
+	if len(nrs.Status.PendingNodes) == 0 && nrs.Status.NodeInProgress == nil {
 		r.EventRecorder.Eventf(
-			nrs, models.Normal, models.DeletionStarted,
-			"Resource is deleted.",
+			nrs, models.Normal, models.UpdatedEvent,
+			"The controller has finished working",
 		)
 		l.Info(
-			"Nodes were reloaded, resource was deleted",
-			"Node Reload spec", nrs.Spec,
+			"The controller has finished working",
+			"completed nodes", nrs.Status.CompletedNodes,
+			"failed nodes", nrs.Status.FailedNodes,
 		)
-		return models.ExitReconcile, nil
+
+		return reconcile.Result{}, nil
 	}
-	patch := nrs.NewPatch()
 
-	if nrs.Status.NodeInProgress.ID == "" {
-		nodeInProgress := &v1beta1.Node{
-			ID: nrs.Spec.Nodes[len(nrs.Spec.Nodes)-1].ID,
-		}
-		nrs.Status.NodeInProgress.ID = nodeInProgress.ID
+	if nrs.Status.NodeInProgress == nil {
+		nodeInProgress := nrs.Status.PendingNodes[0]
```

Comment:
Let's keep the previous indexing, like `nrs.Status.PendingNodes[len(nrs.Status.PendingNodes)-1]`, wdyt?